File: c++/XFile55.cpp
#include "pt4.h"
#include <vector>
#include <fstream>
#include <stdio.h>
using namespace std;

void Solve()
{
    Task("XFile55");
	string out; pt >> out; 
	fstream fout; fout.open(out, ios::binary | ios :: out);
	int n; pt >> n;
	fstream *f = new fstream[n];
	for (int i=0;i<n;i++){
		string s;
		pt>>s;
		f[i].open(s,ios::binary | ios :: in);
		vector <int> v;
		while (f[i].peek() != -1){
		int x;
		f[i].read((char*)&x,sizeof(x));
		v.push_back(x);
		}
		f[i].close();
		int x = v.size();
		fout.write((char*)&x,sizeof(x));
		for(int i=0; i<v.size();i++) {
		int x = v[i];
		fout.write((char*)&x,sizeof(x));
		}
	}
	fout.close();
	delete[] f;
}

File: c#/XFile55.cs
// File: "XFile55"
using PT4;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.IO;

namespace PT4Tasks
{
    public class MyTask : PT
    {
        public static void Solve()
        {
            Task("XFile55");
            string out1 = GetString();
            int n = GetInt();
            BinaryWriter binWriter1 = new BinaryWriter(File.Open(out1, FileMode.Create));

            for(int i=0; i<n;i++){
                string inp = GetString();
                var lenl = new FileInfo(inp).Length;
                lenl /= 4;
                int len = unchecked((int)lenl);
                BinaryReader binReader = new BinaryReader(File.Open(inp, FileMode.Open));
                binWriter1.Write(len);
                for (int j = 0; j< len; j++){
                    int value = binReader.ReadInt32(); 
                    Show(value);
                    binWriter1.Write(value);
                }
                binReader.Close();
            }
            binWriter1.Close();
        }
    }
}
